fifo-sns-sqs-lambda-firehose-s3/main.tf

```diff
@@ -6,7 +6,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 4.0"
+      version = "~> 5.0"
```

Comment:
note: Older version does not support Python 3.12 runtime. So I updated the version to v5.